napari/layers/vectors/vectors.py

```diff
@@ -653,10 +653,14 @@ def _slice_data(
             values, based on how far from the current slice they originate.
         """
         not_disp = list(self._dims_not_displayed)
-        indices = np.array(dims_indices)
+        # We want a numpy array so we can use fancy indexing with the non-displayed
+        # indices, but as dims_indices can (and often/always does) contain slice
+        # objects, the array has dtype=object which is then very slow for the
+        # arithmetic below.
+        not_disp_indices = np.array(dims_indices)[not_disp].astype(float)
```

Comment:
if this variable is used only in the `if` branch then I prefer to move its definition also in this if branch.